taxonomy/api.py

```diff
@@ -0,0 +1,249 @@
+        term_id (int): Term id
+    """
+    get_object_or_404(Term, id=term_id).delete()
+
+
+def get_learning_resources_for_term(term_id):
+    """
+    Get LearningResources using a particular Term
+
+    Args:
+        term_id (int): Term id
+
+    Returns:
+        sequence of LearningResource relating to existing Term
+    """
+    return get_object_or_404(
+        Term,
+        id=term_id
+    ).learning_resources.order_by('id')
+
+
+def get_terms_for_learning_resource(learning_resource_id):
+    """
+    Get Terms using a particular LearningResource
+
+    Args:
+        learning_resource_id (int): Learning object id
+
+    Returns:
+        sequence of Term relating to existing LearningResource
+    """
+    return get_object_or_404(
+        LearningResource, id=learning_resource_id).terms.order_by('label')
+
+
+@transaction.atomic
+def add_term_to_learning_resource(learning_resource_id, term_id):
+    """
+    Add existing Term with id term_id to a particular LearningResource
+    with id learning_resource_id
+
+    Args:
+        learning_resource_id (int): Id for LearningResource
+        term_id (int): Id for Term
+    """
+    learning_resource = get_object_or_404(LearningResource,
+                                          id=learning_resource_id)
+    term = get_object_or_404(Term, id=term_id)
+    learning_resource.terms.add(term)
+
+
+# pylint complains that name has 33 characters when the max is 30
+# pylint: disable=invalid-name
+@transaction.atomic
+def remove_term_from_learning_resource(learning_resource_id, term_id):
+    """
+    Remove existing Term with id term_id from a LearningResource with id
+    learning_resource_id
+
+    Args:
+        learning_resource_id (int): Id for LearningResource
+        term_id (int): Id for Term
+    """
+    term = get_object_or_404(Term, id=term_id)
+    learning_resource = get_object_or_404(LearningResource,
+                                          id=learning_resource_id)
+    term.learning_resources.remove(learning_resource)
+
+
+def get_vocabularies_for_type(learning_resource_type):
+    """
+    Get vocabularies supporting the given learning object type
+
+    Args:
+        learning_resource_type (unicode): The learning resource type
+
+    Returns:
+        sequence of Vocabulary objects supporting the learning resource type
+    """
+    return get_object_or_404(
+        LearningResourceType,
+        name=learning_resource_type
+    ).vocabularies.order_by('name')
+
+
+def get_types_for_vocabulary(vocabulary_id):
+    """
+    Get learning object types supported by a Vocabulary
+
+    Args:
+        vocabulary_id (int): Id for existing Vocabulary
+
+    Returns:
+        list of unicode: sequence of learning object types supported by
+            a LearningResource
+    """
+    learning_resource_types = get_object_or_404(
+        Vocabulary, id=vocabulary_id
+    ).learning_resource_types.order_by('name')
+
+    return (x.name for x in learning_resource_types)
```

Comment:
I was thinking along the lines of maybe someone checking the `len()` of the output and being surprised. I agree generators are nice. That's why I asked if it was intentional, so it could be made more clear, at least with a comment.